point_pattern.py

```diff
@@ -119,7 +139,24 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
+
+    sum_dist = 0
+    for pair_one in points:
+        first = True
+        for pair_two in points:
+            if pair_one == pair_two:
```

Comment:
Check out `enumerate`.  This logic works here, but fails if you have two valid coincident points.  You can solve this by checking positional indices.  If you are at position 0 in both lists, it is the same point.